hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/BenchmarkThroughput.java

```diff
@@ -134,9 +134,9 @@ private void readFile(FileSystem fs,
     InputStream in = fs.open(f);
     byte[] data = new byte[BUFFER_SIZE];
     long val = 0;
-    while (val >= 0) {
+    do {
       val = in.read(data);
-    }
+    } while (val > 0);
```

Comment:
I don't think this is fix is correct. According to the [InputStream.read](762db5bde1c69b54458bea8eb69aa544636c709a) doc, "the total number of bytes read into the buffer, or -1 if there is no more data because the end of the stream has been reached". It is possible that BUFFER_SIZE > 0 and val == 0, in this case, it should continue to read, otherwise it will miss the data. Instead, I think we should check BUFFER_SIZE explicitly.